test/api/AssetTest.ts

```diff
@@ -239,6 +249,107 @@ describe('Asset', function() {
         expect(response.status).to.be.eq(StatusCodes.OK);
         expect(response.data.count).to.be.eq(1);
       });
+
+    });
+
+    describe('Where admin user with asset ', () => {
```

Comment:
You can include the tests in the already existing admin test section.

Reply:
The before section of this describe checks if the keys for ioThink are set. If not, this single describe case fails whereas all the non asset connector based admin test cases are still checked for in the previous describe.

Reply:
I think I will try an alternative of skipping certain testcases if the variables are not set like [here](https://stackoverflow.com/questions/32723167/how-to-programmatically-skip-a-test-in-mocha). This way, I can integrate all of them into one single admin test cases.

Reply:
I think its better to leave it in a separate describe instead of adding it to admin cases. This is because, in 'admin only' case, we are not creating an asset in the before section and the test is written to check if asset creation is supported for an admin. Here we are checking if an admin with existing asset can connect to a connector. This way, we need an asset created for every test case in this describe due to which it is added in the before section but not in the previous 'admin only' describe. 
One alternative is to create an asset in the first asset connector test case but this will create a dependency between the test. Then one can't run the test cases independently and need to run all of them so that the asset is created. With the current way, the before section is executed before a single or all test case that is run.